services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -481,6 +476,51 @@ public LinkedAccountsResource linkedAccounts() {
         return new LinkedAccountsResource(session, request, client, auth, event, user);
     }
 
+    @Path("/applications")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public Response applications() {
+        checkAccountApiEnabled();
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_APPLICATIONS);
+
+        Set<ClientModel> clients = new HashSet<>();
+        List<String> inUseClients = new LinkedList<String>();
+        List<UserSessionModel> sessions = session.sessions().getUserSessions(realm, user);
+        for(UserSessionModel s : sessions) {
+            for (AuthenticatedClientSessionModel a : s.getAuthenticatedClientSessions().values()) {
+                ClientModel client = a.getClient();
+                clients.add(client);
+                inUseClients.add(client.getClientId());
+            }
+        }
+
+        List<String> offlineClients = new LinkedList<String>();
+        List<UserSessionModel> offlineSessions = session.sessions().getOfflineUserSessions(realm, user);
```

Comment:
I'm not 100% sure, but I think getUserSessions returns offline sessions as well, and you can check if it's a offline session with isOffline. That would prevent doing to queries for sessions.

Reply:
Unfortunately this change isn't possible. getUserSessions does return all of the sessions but it sets offline = false for all of them. The only way to know if a session is an offline session is to call getOfflineUserSessions.

We should probably change the behaviour of getUserSessions but I think that is beyond the scope of this PR.